src/snowflake/snowpark/table.py

```diff
@@ -270,27 +270,36 @@ def __init__(
         self,
         table_name: str,
         session: Optional["snowflake.snowpark.session.Session"] = None,
+        is_generated_temp_table: bool = False,
     ) -> None:
-        super().__init__(
```

Comment:
The previous way is actually not correct because setting `_plan` to UnresolvedRelation even if sql simplifier is enabled. I can also put out a separate PR for it if necessary.

Reply:
Yes, if it is an independent problem, let's do it in a different pr. However, i don't think i am quite getting the problem here, can you be more specific here?

Reply:
Yes, remove the refactor for now. The reason is that in DataFrame's constructor, it will call session._analyzer.resolve again.